src/main/kotlin/com/qupaya/brevo/NewsletterRegistrationEventListenerProvider.kt

```diff
@@ -66,30 +67,26 @@ class NewsletterRegistrationEventListenerProvider(
          * using the form data POST request ensures that handling like
          * success pages can be controlled inside Brevo.
          */
-        val http = HttpClient.newHttpClient()
-        val request = HttpRequest.newBuilder()
-            .uri(URI.create(brevoFormLink))
-            .POST(HttpRequest.BodyPublishers.ofString(getFormDataAsString(formData)))
-            .header("content-type", "application/x-www-form-urlencoded")
-            .build()
+        val httpPost = HttpPost(brevoFormLink)
+        httpPost.entity = StringEntity(getFormDataAsString(formData))
+        httpPost.setHeader("content-type", "application/x-www-form-urlencoded")
 
-        for (waitTime in RETRY_WAIT_DURATIONS) {
-            Thread.sleep(waitTime.toMillis())
-
-            try {
-                val response = http.send(request, HttpResponse.BodyHandlers.ofString())
-
-                if (response.statusCode() < HTTP_ERROR_CODES_START) { // success
-                    break
+        try {
+            HttpClientBuilder.create()
+                .setRetryHandler(DefaultHttpRequestRetryHandler())
+                .setRedirectStrategy(LaxRedirectStrategy()).build()
+                .use { http ->
+                    val response = http.execute(httpPost)
+                    if (response.statusLine.statusCode > HTTP_ERROR_CODES_START) {
+                        LOG.error("Brevo newsletter subscription request response: ${response.statusLine.statusCode}")
+                        LOG.error(EntityUtils.toString(response.entity))
+                    }
                 }
 
-                LOG.error("Brevo newsletter subscription request response: ${response.statusCode()}")
-                LOG.error(response.body())
-            } catch (ex: IOException) {
-                LOG.error("IO exception while sending newsletter subscription request for user with ID $userId.", ex)
-            } catch (ex: InterruptedException) {
-                LOG.error("Interruption while sending newsletter subscription request for user with ID $userId.", ex)
-            }
+        } catch (ex: IOException) {
+            LOG.error("IO exception while sending newsletter subscription request for user with ID $userId.", ex)
```

Comment:
probably we will need some json logging in the future, so that the logs are searchable in Grafana